### PR TITLE
Fix postgres datasource devservice not working with rancher-desktop on mac arm

### LINKED ISSUE #25682: Dev Services for Postgresql not working with Rancher
### Describe the bug

Last week I got a new laptop - a 2021 MacBook M1Pro. Since then whenever I try to use Dev Services for Postgresql (either in Quarkus Dev mode or when running tests) it seems to be broken with some kind of race condition between the starting of the Postgres container & the creation of the username/password). When Hibernate kicks in to create the tables its getting an authentication error, as if the user hasn’t yet been created. I’m hoping my new machine isn’t too fast!

### Expected behavior

I would expect the dev service for postgresql to work.

### Actual behavior

It does not

### How to Reproduce?

I can reproduce this pretty regularly. I put together a repo on GitHub [1] detailing what I’m seeing (it’s a pretty basic hello world). I also recorded a 2 minute video [2] showing what I’m seeing. The [README file in the repo](https://github.com/edeandrea/quarkus-devservice-postgresql/blob/main/README.md) has a very detailed explanation of the problem & what I'm seeing.

[1] https://github.com/edeandrea/quarkus-devservice-postgresql
[2] https://github.com/edeandrea/quarkus-devservice-postgresql/raw/main/Dev-Services-postgres-failing.mov

### Output of `uname -a` or `ver`

```
Darwin edeandrea-m1pro 21.5.0 Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:37 PDT 2022; root:xnu-8020.121.3~4/RELEASE_ARM64_T6000 arm64
```

### Output of `java -version`

```
openjdk version "17.0.3" 2022-04-19
OpenJDK Runtime Environment Temurin-17.0.3+7 (build 17.0.3+7)
OpenJDK 64-Bit Server VM Temurin-17.0.3+7 (build 17.0.3+7, mixed mode)
```

### Quarkus version or git rev

2.9.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

```
Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)
Maven home: /Users/edeandre/.m2/wrapper/dists/apache-maven-3.8.4-bin/52ccbt68d252mdldqsfsn03jlf/apache-maven-3.8.4
Java version: 17.0.3, vendor: Eclipse Adoptium, runtime: /Users/edeandre/.sdkman/candidates/java/17.0.3-tem
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "12.4", arch: "aarch64", family: "mac"
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by bpasson**
Fixes #25682 postgres datasource devservice not working with rancher-desktop on mac arm architecture.

@gastaldi @gsmet I tested this on a mac arm m1 and it works. I don't have the option to test other platforms. We probably need to make sure it works on all of them.

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]

File: extensions/devservices/postgresql/src/main/java/io/quarkus/devservices/postgresql/deployment/PostgresqlDevServicesProcessor.java [hunk 1]
```diff
@@ -99,16 +101,23 @@ public QuarkusPostgreSQLContainer(Optional<String> imageName, OptionalInt fixedE
...
             // This custom wait strategy will work fine regardless of the state of the Postgresql database.
             // More information in the issue ticket in Testcontainers.
-            this.waitStrategy = new LogMessageWaitStrategy()
-                    .withRegEx("(" + READY_REGEX + ")?(" + SKIPPING_INITIALIZATION_REGEX + ")?")
-                    .withTimes(2)
+
+            // Added Wait.forListeningPort() for https://github.com/quarkusio/quarkus/issues/25682
+            // as suggested by https://github.com/testcontainers/testcontainers-java/pull/6309
+            this.waitStrategy = new WaitAllStrategy()
+                    .withStrategy(new LogMessageWaitStrategy()
```
Comment (gastaldi) [inline extensions/devservices/postgresql/src/main/java/io/quarkus/devservices/postgresql/deployment/PostgresqlDevServicesProcessor.java:117]: Maybe use `Wait.forLogMessage` here for simplification purposes?

Comment (bpasson) [reply extensions/devservices/postgresql/src/main/java/io/quarkus/devservices/postgresql/deployment/PostgresqlDevServicesProcessor.java]: Done.

**Comment by bpasson** [comment]
@gastaldi did you see my comment about testing on different platforms and architectures?

**Comment by gastaldi** [comment]
@bpasson I did, CI should hopefully run in Linux and Windows, which should hopefully be enough in this case

**Review by gsmet — APPROVED** [review]

**Pull request merged**